types/test/types/rpc.ts

```diff
@@ -306,6 +306,27 @@ class TestObject extends DurableObject {
   }
 }
 
+class TestAlarmObject extends DurableObject {
+  // Can declare alarm method consuming optional alarmInfo parameter
+  async alarm(alarmInfo?: AlarmInvocationInfo) {
+    if (alarmInfo !== undefined) {
```

Comment:
FWIW, the current version of the runtime always passes this `alarmInfo` parameter, but early implementations did not, and it's likely that almost all Durable Object `alarm()` implementations in the wild do not currently declare the parameter.  I think that means that the parameter needs to be declared `alarmInfo?` for backwards compatibility, although it is a bit unfortunate that this requires TS DOs to do this check for `undefined` before using the parameter.

I'm not sure if there is a better way to declare the method in the TS `rpc.d.ts` file such that it allows class implementers to either consume a (non-undefined) `alarmInfo` parameter or to omit it entirely.

Probably not a big deal in the common case, though, since most implementers probably won't be writing an `alarm()` handler that needs the info in the parameter.

Reply:
Dumb question: is there a way to set the `alarmInfo` as always passed as of a particular compatibility date?

Reply:
Not dumb -- I was wondering myself, but am not familiar with how we are doing versioning and backwards compatibility of TypeScript declarations.

I assume the purpose of the various `types/generated-snapshot/*` files might be to allow for compatibility-breaking changes in types, but I'm not sure if there is a correspondence to compatibility dates.  I think marking the parameter as mandatory would not be backwards compatible, in the sense that it could cause TS to issue new errors or warnings for existing worker code -- either for `alarm()` handler definitions, or in cases where worker code tries to invoke `alarm()` handlers directly.

I suspect this feature will be infrequently used enough that it might not be worth breaking backwards compatibility for, by itself -- but if there comes a time when we're breaking backwards compatibility anyway, that might be a convenient time to make the parameter mandatory?